beta-src/src/components/controllers/WDUnitController.tsx

```diff
@@ -11,26 +8,27 @@ import WDArmyIcon, {
   ARMY_RAW_ICON_WIDTH,
   ARMY_RAW_ICON_HEIGHT,
 } from "../ui/units/WDArmyIcon";
+import { Unit } from "../../utils/map/getUnits";
+import UIState from "../../enums/UIState";
+import { makeSVGDrawAsUnsavedAnimateElementGentle } from "../../utils/map/drawArrowFunctional";
 
 interface UnitControllerProps {
-  meta: GameIconProps["meta"];
-  type: GameIconProps["type"];
-  iconState: GameIconProps["iconState"];
+  unit: Unit;
+  unitState: UIState;
   unitWidth: number;
   unitHeight: number;
 }
 
 const WDUnitController: React.FC<UnitControllerProps> = function ({
-  meta,
-  type,
-  iconState,
+  unit,
+  unitState,
   unitWidth,
   unitHeight,
 }): React.ReactElement {
   const iconWidth =
-    type === "Fleet" ? FLEET_RAW_ICON_WIDTH : ARMY_RAW_ICON_WIDTH;
+    unit.unit.type === "Fleet" ? FLEET_RAW_ICON_WIDTH : ARMY_RAW_ICON_WIDTH;
```

Comment:
Is there a way to write this without needing to access it with `unit.unit`? I feel like there are two different meanings of `unit` here, and it's not obvious what the difference is during a skim read.

As an aside, knowing how to calculate the width and height doesn't feel like the responsibility of a controller - it might read better as a function that returns `{ iconWidth, iconHeight }`